lib/tasks/test_unit.rake

```diff
@@ -1,7 +1,43 @@
+#--
+# Yast rake
+#
+# Copyright (C) 2018 SUSE LLC
+#   This library is free software; you can redistribute it and/or modify
+# it only under the terms of version 2.1 of the GNU Lesser General Public
+# License as published by the Free Software Foundation.
+#
+#   This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+# details.
+#++
+
 namespace :test do
   desc "Runs unit tests."
   task "unit" do
-    files = Dir["**/test/**/*_{spec,test}.rb"]
-    sh "rspec --color --format doc '#{files.join("' '")}'" unless files.empty?
+    files = Dir["**/test/**/*_{spec,test}.rb"].join("' '")
+    if !files.empty?
+      # use parallel_tests if configured, allow changing it via environment
+      if (File.exist?(".rspec_parallel") && ENV["PARALLEL_TESTS"] != "0") ||
+          ENV["PARALLEL_TESTS"] == "1"
+
+        # pass custom parameters to parallel_rspec if needed,
+        # e.g. the number of CPUs to use
+        sh "parallel_rspec --verbose #{ENV["PARALLEL_TESTS_OPTIONS"]} '#{files}'"
```

Comment:
maybe we should unset `ENV["COVERAGE"]` here so it is not send multiple times?

Reply:
You should fix the test helper to not submit the coverage data when using parallel tests. Overriding the defaults with `PARALLEL_TESTS=1` should be done only when testing locally, not in Travis.

Unfortunately we cannot unset `COVERAGE` as that would disable the code coverage completely, then there would be nothing to send to the server. See this example: https://github.com/yast/yast-bootloader/blob/master/test/test_helper.rb#L28

We could unset `TRAVIS` variable to achieve that, but that does not look like as a good idea to me. We might want to skip some tests in Travis (because of some system incompatibility as it is running in Docker in Ubuntu OS or for what ever else reason). The Travis detection then would be broken.

So I'd keep it as it is.